internal/version/release_version.go

```diff
@@ -40,15 +39,30 @@ type releaseVersionFetcher struct {
 	ctx context.Context
 }
 
-// LatestVersion encapsulates the logic to manage different cloud providers.
-func (s *releaseVersionFetcher) LatestVersion() (*ReleaseInformation, error) {
+// LatestWithCriteria retrieves the first release version that matches the criteria. We assume that ListReleases returns releases sorted by created_at value.
+func (s *releaseVersionFetcher) LatestWithCriteria(n int, matchCriteria func(tag string, tool string) bool, toolName string) (*github.RepositoryRelease, error) {
+	var page = 1
+	var pageSize = n
+
+	startTime := time.Now()
 	client := github.NewClient(nil)
-	release, _, err := client.Repositories.GetLatestRelease(s.ctx, "mongodb", "mongocli")
-	if err != nil {
-		return nil, err
+
+	for {
+		releases, _, err := client.Repositories.ListReleases(s.ctx, owner, mongoCLI, &github.ListOptions{PerPage: pageSize, Page: page})
+		if err != nil {
+			return nil, err
+		}
+		// Returns as soon as criteria is matched
+		for i := 0; i < len(releases); i++ {
+			release := releases[i]
+			if !release.GetDraft() && !release.GetPrerelease() && matchCriteria(release.GetTagName(), toolName) {
```

Comment:
from the api docs https://docs.github.com/en/rest/reference/releases#list-releases
> Only users with push access will receive listings for draft releases.

since this is an anonymous api call you'll never get a draft release

Reply:
thanks, removing draft check